core/Cargo.toml

```diff
@@ -28,7 +29,6 @@ rustc-hash = { version = "1", optional = true }
 rand = { version = "0.8", optional = true }
 soketto = { version = "0.7.1", optional = true }
 parking_lot = { version = "0.12", optional = true }
-tokio = { version = "1.16", optional = true }
```

Comment:
Is there a downside to tokio becoming non-optional?

Reply:
bigger dependency tree :(

hmm, I think it's because of `TokioSendError` impl

Reply:
I guess I'm wondering; is it worth the extra dependencies etc to take this approach rather than the original one of an extra Notify thing?

I think I like this approach overall but would be good to get your opinion too on which you prefer, now you've been both!

Reply:
Have picked the original approach and utilized `futures_channel` instead of `tokio` to avoid having an extra dependency. Mainly because the tokio's sender channel cannot be closed directly. It is only closed when all senders are dropped, and we have it cloned in the subscriptions.

For now, utilizing this approach seems like a good tradeoff between dependencies/code to maintain. We can in the future change it under the hood, as it would likely not imply any breaking changes.